gr-filter/lib/pfb_arb_resampler.cc

```diff
@@ -197,24 +200,34 @@ float pfb_arb_resampler_ccf::phase_offset(float freq, float fs)
     return -adj * d_est_phase_change;
 }
 
+void pfb_arb_resampler_ccf::run_thread(int filter_index, gr_complex* input)
+{
+    d_o0 = d_filters[filter_index]->filter(input);
+}
+
 int pfb_arb_resampler_ccf::filter(gr_complex* output,
                                   gr_complex* input,
                                   int n_to_read,
                                   int& n_read)
 {
+    gr::thread::scoped_lock guard(d_mutex);
+
     int i_out = 0, i_in = 0;
     unsigned int j = d_last_filter;
-    ;
-    gr_complex o0, o1;
+    gr_complex o1;
 
     while (i_in < n_to_read) {
         // start j by wrapping around mod the number of channels
         while (j < d_int_rate) {
             // Take the current filter and derivative filter output
-            o0 = d_filters[j]->filter(&input[i_in]);
+            filter_thread = new boost::thread(
```

Comment:
I strongly doubt this will increase performance. You create a thread for every single sample to be processed. Thread creation alone will increase latency (however fast it is). Also, this requires free CPU time to be available the very moment the thread is created. If this is not the case, latency will increase even further.

Reply:
I agree, I think that's what I ran into.  I have another way I've done it without creating the threads every call that should alleviate that.  I'm going to try to see if the performance stays.